samples/drm/app/main.js

```diff
@@ -0,0 +1,374 @@
+'use strict';
+
+angular.module('DashSourcesService', ['ngResource']).
+    factory('Sources', function($resource){
+        return $resource('app/sources.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashNotesService', ['ngResource']).
+    factory('Notes', function($resource){
+        return $resource('app/notes.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashContributorsService', ['ngResource']).
+    factory('Contributors', function($resource){
+        return $resource('app/contributors.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashPlayerLibrariesService', ['ngResource']).
+    factory('PlayerLibraries', function($resource){
+        return $resource('app/player_libraries.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashShowcaseLibrariesService', ['ngResource']).
+    factory('ShowcaseLibraries', function($resource){
+        return $resource('app/showcase_libraries.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+var app = angular.module('DashPlayer', [
+    'DashSourcesService',
+    'DashNotesService',
+    'DashContributorsService',
+    'DashPlayerLibrariesService',
+    'DashShowcaseLibrariesService'
+]);
+
+app.controller('DashController', function($scope, Sources, Notes, Contributors, PlayerLibraries, ShowcaseLibraries) {
+    var player,
+        video,
+        context;
+
+    $scope.drmData = [];
+
+    $scope.safeApply = function(fn) {
+        var phase = this.$root.$$phase;
+        if(phase == '$apply' || phase == '$digest')
+            this.$eval(fn);
+        else
+            this.$apply(fn);
+    };
+
+    ////////////////////////////////////////
+    //
+    // Error Handling
+    //
+    ////////////////////////////////////////
+
+    function onError(e) {
+
+    }
+
+    ////////////////////////////////////////
+    //
+    // Player Setup
+    //
+    ////////////////////////////////////////
+
+    video = document.querySelector(".dash-video-player video");
+    context = new Dash.di.DashContext();
+    player = new MediaPlayer(context);
+    $scope.version = player.getVersion();
+
+    player.startup();
+    player.addEventListener(MediaPlayer.events.ERROR, onError.bind(this));
+
+    player.attachView(video);
+    player.setAutoPlay(true);
+
+    var system = player.getSystem(),
```

Comment:
@greg80303, I have been thinking about this and I still can't quite understand why we need to use all these classes externally. I mean, why can't we just move fetching license logic into MediaPlayer.fetchLicense method. We could even move manifest loading logic in a separate MediaPlayler.fetchMpd function. My reasoning is that using internal classes externally we expose too much implementation details, which is not good in case we will decide to change some of those classes. Besides, it requires knowing of a lot of methods of those classes, e.g. users have to know how to use manifest loader, how to use DashAdapter (which will probably be removed from the project soon) etc.

This is just my personal opinion, so it would be great to know other opinions to find a compromise.